pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -147,17 +147,15 @@ func (s *Scheduler) getSchedules(ctx context.Context, constraints *v1alpha5.Cons
 }
 
 func (s *Scheduler) getDaemons(ctx context.Context, constraints *v1alpha5.Constraints) ([]*v1.Pod, error) {
-	// 1. Get DaemonSets
 	daemonSetList := &appsv1.DaemonSetList{}
 	if err := s.KubeClient.List(ctx, daemonSetList); err != nil {
 		return nil, fmt.Errorf("listing daemonsets, %w", err)
 	}
-
-	// 2. filter DaemonSets to include those that will schedule on this node
+	// Include daemonsets that will schedule on this node
 	pods := []*v1.Pod{}
 	for _, daemonSet := range daemonSetList.Items {
 		pod := &v1.Pod{Spec: daemonSet.Spec.Template.Spec}
-		if constraints.Supports(pod) != nil {
+		if constraints.Supports(pod) == nil {
```

Comment:
I think part of the issue here is that the name `Supports` (which I think reads most naturally as "supports?") suggests a boolean (true/false)? Is there some reason this has to return a nullable value?

Reply:
This is going to change a bit soon. I want to cover this in a refactor, but for now, I've renamed to ValidatePod